packages/radio-group/src/vaadin-radio-group.js

```diff
@@ -73,510 +85,380 @@ class RadioGroupElement extends ThemableMixin(DirMixin(PolymerElement)) {
     });
   }
 
-  /** @private */
-  get _radioButtons() {
-    return this._filterRadioButtons(this.querySelectorAll('*'));
-  }
-
   /**
-   * @param {boolean} focused
-   * @protected
+   * @param {!HTMLCollection | !HTMLElement[]} nodes
+   * @return {!RadioButton[]}
+   * @private
    */
-  _setFocused(focused) {
-    if (focused) {
-      this.setAttribute('focused', '');
-    } else {
-      this.removeAttribute('focused');
-    }
-  }
-
-  /** @private */
-  _filterRadioButtons(nodes) {
+  __filterRadioButtons(nodes) {
     return Array.from(nodes).filter((child) => child instanceof RadioButton);
   }
 
-  /** @private */
-  _disabledChanged(disabled) {
-    this.setAttribute('aria-disabled', disabled);
-    this._updateDisableButtons();
-  }
-
-  /** @private */
-  _updateDisableButtons() {
-    this._radioButtons.forEach((button) => {
-      if (this.disabled) {
-        button.disabled = true;
-      } else if (this.readonly) {
-        // it's not possible to set readonly to radio buttons, but we can
-        // unchecked ones instead.
-        button.disabled = button !== this._checkedButton && this.readonly;
-      } else {
-        button.disabled = false;
-      }
-    });
+  /**
+   * A collection of the group's radio buttons.
+   *
+   * @type {!RadioButton[]}
+   * @private
+   */
+  get __radioButtons() {
+    return this.__filterRadioButtons(this.children);
   }
 
-  /** @private */
-  _readonlyChanged(newV, oldV) {
-    (newV || oldV) && this._updateDisableButtons();
+  /**
+   * A currently selected radio button.
+   *
+   * @type {!RadioButton | undefined}
+   * @private
+   */
+  get __selectedRadioButton() {
+    return this.__radioButtons.find((radioButton) => radioButton.checked);
   }
 
-  /** @private */
-  _addListeners() {
-    this.addEventListener('keydown', (e) => {
-      // if e.target is vaadin-radio-group then assign to checkedRadioButton currently checked radio button
-      let checkedRadioButton;
-
-      if (e.target === this) {
-        checkedRadioButton = this._checkedButton;
-      } else {
-        checkedRadioButton = e.composedPath().find((element) => {
-          return element instanceof RadioButton;
-        });
-      }
-      const horizontalRTL = this.getAttribute('dir') === 'rtl' && this.theme !== 'vertical';
-
-      // LEFT, UP - select previous radio button
-      if (e.keyCode === 37 || e.keyCode === 38) {
-        e.preventDefault();
-        this._selectIncButton(horizontalRTL, checkedRadioButton);
-      }
-
-      // RIGHT, DOWN - select next radio button
-      if (e.keyCode === 39 || e.keyCode === 40) {
-        e.preventDefault();
-        this._selectIncButton(!horizontalRTL, checkedRadioButton);
-      }
-    });
-
-    this.addEventListener('focusin', () => this._setFocused(this._containsFocus()));
-
-    this.addEventListener('focusout', () => {
-      this.validate();
-      this._setFocused(false);
-    });
+  /**
+   * @type {!boolean}
```

Comment:
We only use `!` in JSDoc for types like `Array` or `RadioButton`, but not for primitive types.

```suggestion
   * @return {boolean}
```

Reply:
Got it, done.